app/src/main/java/com/github/steroidteam/todolist/view/MapsActivity.java

```diff
@@ -0,0 +1,261 @@
+                location.substring(0, 1).toUpperCase() + location.substring(1);
+        if (marker != null) {
+            marker.setPosition(latLng);
+            marker.setTitle(locationWithUppercase);
+        } else {
+            marker =
+                    map.addMarker(
+                            new MarkerOptions().position(latLng).title(locationWithUppercase));
+        }
+        map.animateCamera(CameraUpdateFactory.newLatLngZoom(latLng, 10));
+    }
+
+    /**
+     * Manipulates the map when it's available. This callback is triggered when the map is ready to
+     * be used.
+     */
+    @Override
+    public void onMapReady(GoogleMap map) {
+        this.map = map;
+        // Prompt the user for permission.
+        getLocationPermission();
+
+        // Turn on the My Location layer and the related control on the map.
+        updateLocationUI();
+
+        // Get the current location of the device and set the position of the map.
+        getDeviceLocation();
+    }
+
+    /** Gets the current location of the device, and positions the map's camera. */
+    private void getDeviceLocation() {
+        /*
+         * Get the best and most recent location of the device, which may be null in rare
+         * cases when a location is not available.
+         */
+        try {
+            if (locationPermissionGranted) {
+                Task<Location> locationResult = fusedLocationProviderClient.getLastLocation();
+                locationResult.addOnCompleteListener(this, task -> onCompleteDeviceLocation(task));
+            }
+        } catch (SecurityException e) {
+            Log.e("Exception: %s", e.getMessage(), e);
+        }
+    }
+
+    private void onCompleteDeviceLocation(@NonNull Task<Location> task) {
+        if (task.isSuccessful()) {
+            // Set the map's camera position to the current location of the device.
+            lastKnownLocation = task.getResult();
+            deviceLocationMarker(lastKnownLocation);
+        } else {
+            Log.d(TAG, "Current location is null. Using defaults.");
+            Log.e(TAG, "Exception: %s", task.getException());
+            map.moveCamera(CameraUpdateFactory.newLatLngZoom(defaultLocation, DEFAULT_ZOOM));
+            map.getUiSettings().setMyLocationButtonEnabled(false);
+        }
+    }
+
+    public void deviceLocationMarker(Location location) {
+        if (location != null) {
+            deviceLocationMarkerLocNotNull(location);
+        } else {
+            deviceLocationMarkerLocIsNull();
+        }
+    }
+
+    private void deviceLocationMarkerLocNotNull(Location location) {
+        map.moveCamera(
+                CameraUpdateFactory.newLatLngZoom(
+                        new LatLng(location.getLatitude(), location.getLongitude()), DEFAULT_ZOOM));
+        if (marker != null) {
+            marker.setPosition(new LatLng(location.getLatitude(), location.getLongitude()));
+            marker.setTitle("I'm here");
```

Comment:
Also for a string like this maybe you should extract it and add a resource in strings.xml.
Then you can recover it in the activity with `getString(R.string.your_string_name);`

Reply:
Yes you're completely right ! Thanks a lot for pointing this out I've corrected it right now